visitor.php

```diff
@@ -1000,66 +1007,87 @@ private static function isOptional(string $description): bool
+            // it's not return type void.
+            if (
+                $this->nodeFinder->findFirst(
+                    $return,
+                    static function (Node $node): bool {
+                        return isset($node->expr);
+                    }
+                ) !== null
+            ) {
+                return '';
+            }
+            // If there is no return statement that is not void,
+            // it's return type void.
+            return 'void';
         }
 
-        // If wp_die is called with 3rd parameter, we need additional checks.
-        $argValue = $args[2]->value;
-        if ($argValue instanceof Variable) {
-            return false;
-        }
-        if ($argValue instanceof Array_) {
-            foreach ($argValue->items as $item ) {
-                if ($item instanceof ArrayItem && $item->key instanceof String_ && $item->key->value === 'exit') {
-                    return false;
+        // Check for never return type.
+        foreach ($node->stmts as $stmt) {
+            if (!($stmt instanceof Expression)) {
+                continue;
+            }
+            // If a first level statement is exit/die, it's return type never.
+            if ($stmt->expr instanceof Exit_) {
+                return 'never';
+            }
+            if (!($stmt->expr instanceof FuncCall) || !($stmt->expr->name instanceof Name)) {
+                continue;
+            }
+            $name = $stmt->expr->name;
+            // If a first level statement is a call to wp_send_json(_success/error),
+            // it's return type never.
+            if ($name instanceof Name && strpos($name->toString(), 'wp_send_json') === 0) {
+                return 'never';
+            }
+            // Skip all functions but wp_die().
+            if (strpos($name->toString(), 'wp_die') !== 0) {
+                continue;
+            }
+            $args = $stmt->expr->getArgs();
+             // If wp_die is called without 3rd parameter, it's return type never.
+            if (count($args) < 3) {
+                return 'never';
+            }
+            // If wp_die is called with 3rd parameter, we need additional checks.
+            $argValue = $args[2]->value;
+            if (!($argValue instanceof Array_)) {
+                continue;
+            }
+            foreach ($argValue->items as $item) {
+                if (!($item instanceof ArrayItem && $item->key instanceof String_ && $item->key->value === 'exit')) {
+                    continue;
+                }
+                if (
+                    ($item->value instanceof Node\Expr\ConstFetch && strtolower($item->value->name->toString()) === 'true') ||
+                    ($item->value instanceof Node\Scalar\LNumber && $item->value->value === 1) ||
+                    ($item->value instanceof Node\Scalar\String_ && $item->value->value !== '' && $item->value->value !== '0')
```

Comment:
```suggestion
                    ($item->value instanceof Node\Expr\ConstFetch && strtolower($item->value->name->toString()) === 'true')
                    || ($item->value instanceof Node\Scalar\LNumber && $item->value->value === 1)
                    || ($item->value instanceof Node\Scalar\String_ && $item->value->value !== '' && $item->value->value !== '0')
```

So we won't stand on our head.